src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java

```diff
@@ -13,32 +14,40 @@
     private final AttachDocsToSupplementaryEvidence evidenceAttacher;
     private final CreateExceptionRecord exceptionRecordCreator;
     private final CaseFinder caseFinder;
+    private final PaymentsProcessor paymentsProcessor;
 
     public EnvelopeHandler(
         AttachDocsToSupplementaryEvidence evidenceAttacher,
         CreateExceptionRecord exceptionRecordCreator,
-        CaseFinder caseFinder
+        CaseFinder caseFinder,
+        PaymentsProcessor paymentsProcessor
     ) {
         this.evidenceAttacher = evidenceAttacher;
         this.exceptionRecordCreator = exceptionRecordCreator;
         this.caseFinder = caseFinder;
+        this.paymentsProcessor = paymentsProcessor;
     }
 
     public void handleEnvelope(Envelope envelope) {
         switch (envelope.classification) {
             case SUPPLEMENTARY_EVIDENCE:
-                Optional<CaseDetails> caseDetails = caseFinder.findCase(envelope);
+                Optional<CaseDetails> caseDetailsFound = caseFinder.findCase(envelope);
 
-                if (caseDetails.isPresent()) {
-                    evidenceAttacher.attach(envelope, caseDetails.get());
+                if (caseDetailsFound.isPresent()) {
+                    evidenceAttacher.attach(envelope, caseDetailsFound.get());
                 } else {
                     exceptionRecordCreator.tryCreateFrom(envelope);
                 }
 
                 break;
             case EXCEPTION:
-            case NEW_APPLICATION:
                 exceptionRecordCreator.tryCreateFrom(envelope);
+                break;
+            case NEW_APPLICATION:
+                Long ccdId = exceptionRecordCreator.tryCreateFrom(envelope);
+
+                paymentsProcessor.processPayments(envelope, ccdId, true);
```

Comment:
shouldn't this happen for EXCEPTION as well?

Reply:
No, payments are possible only for NEW_APPLICATION, not EXCEPTION

Reply:
ok